geomesa-core/src/main/scala/org/locationtech/geomesa/core/iterators/AttributeIndexIterator.scala

```diff
@@ -56,16 +64,44 @@ class AttributeIndexIterator
     // if we're retrieving the attribute, we need the class in order to decode it
     attributeType = Option(options.get(GEOMESA_ITERATORS_ATTRIBUTE_NAME))
         .flatMap(n => Option(featureType.getDescriptor(n)))
+    dtgIndex = index.getDtgFieldName(featureType).map(featureType.indexOf(_))
+    val coverage = Option(options.get(GEOMESA_ITERATORS_ATTRIBUTE_COVERAGE)).map(IndexCoverage.withName)
+        .getOrElse(IndexCoverage.JOIN)
+    setTopFunction = coverage match {
+      case IndexCoverage.FULL => setTopFullCoverage
+      case IndexCoverage.JOIN => setTopJoinCoverage
+    }
   }
 
-  override def setTopConditionally() {
+  override def setTopConditionally() = setTopFunction()
+
+  /**
+   * Each value is the fully encoded simple feature
+   */
+  def setTopFullCoverage(): Unit = {
+    val dataValue = source.getTopValue
+    val sf = featureDecoder.decode(dataValue.get)
+    val meetsStFilter = stFilter.forall(fn => fn(sf.getDefaultGeometry.asInstanceOf[Geometry],
```

Comment:
the reason I didn't was b/c stFilter is an option, so it will only be evaluated once, and I wanted to avoid the overhead of creating the vals outside the option if the option was None. I could make them lazy, but there's still some overhead involved...
These methods have actually been completely replaced in some pending work I haven't put up for PRs yet.